### PR TITLE
Allow the user to recover from a lost password

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://github.com/hughsie/lvfs-website/issues/141

### CODE REVIEW

File: app/templates/email-recover-password.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Dear {{ user.display_name }},
+
+Your password has been reset Please click on the following link to log in to the LVFS:
+
+{{ url_for('.index', _external=True) }}
+
+Your credidentials have been set as:
```
Comment (superm1) [inline app/templates/email-recover-password.txt]: spelling error

File: app/views_user.py [hunk 2]
```diff
@@ -223,6 +223,65 @@ def user_modify_by_admin(user_id):
...
+        return _error_permission_denied('Unable to recover user as no username')
+
+    # check exists
+    username = request.form['username']
+    user = db.session.query(User).filter(User.username == username).first()
+    if not user:
+        flash('Unable to recover password as no username %s found' % username, 'warning')
+        return redirect(url_for('.index'), 302)
+
+    # password is stored hashed
```
Comment (superm1) [inline app/views_user.py]: I think there needs to be a check in this flow if their email is configured as local or oauth and abort if they're oauth.

File: app/templates/email-recover-password.txt (hunk 1 above)
Comment (hughsie) [reply app/templates/email-recover-password.txt]: Fixed, thanks.

File: app/views_user.py (hunk 2 above)
Comment (hughsie) [reply app/views_user.py]: ...and if they're a robot, or a locked account. Fixed, thanks.

**Comment by superm1** [comment]
Flow looks good to me now, but I think it would be good to include a test suite test for this if possible since it's a multi step process that's potentially fragile.  I think test suite test could avoid sending/receiving parsing the email and just snoop the token though.

**Pull request merged**